File: Modul 6/soal_4/PRAK604-2410817310003-Muhammad_Guntur_Ricky_Adhitya.c
#include <stdio.h>
#include <string.h>

void kode_shikamaru(char bahasa[], char kode[]){
    if(strlen(bahasa) != strlen(kode)){
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return;
    }
    int pagar = 0, bintang = 0;
    int panjang = strlen(bahasa);
    for(int i = 0; i < panjang; ++i){
        if((bahasa[i] == kode[i]) && (bahasa[i] == ' ')){
            printf(" ");
        }
        else if(bahasa[i] == kode[i]){
            printf("*");
            bintang++;
        }else{
            printf("#");
            pagar++;
        }
    }
    printf("\n");
    printf("* = %d\n", bintang);
    printf("# = %d\n", pagar);

    if(bintang >= pagar)printf("Pesan Asli\n");
    else printf("Pesan Palsu\n");
}

int main(){
    char bahasa[100];
    char kode[100];
    scanf("%[^\n]s", &bahasa);
    getchar();
    scanf("%[^\n]s", &kode);
    getchar();

    kode_shikamaru(bahasa, kode);

    return 0;
}

File: Modul 6/soal_4/PRAK604-2410817310003-Muhammad_Guntur_Ricky_Adhitya.py
def kode_shikamaru(bahasa, kode):
    if(len(bahasa) != len(kode)):
        print("Panjang kalimat berbeda, pesan palsu")
        exit()

    pagar = 0
    bintang = 0
    for i in range(0, len(bahasa)):
        if (bahasa[i] == kode[i]) and bahasa[i] == " ":
            print(end=" ")
        elif (bahasa[i] == kode[i]):
            print("*", end="")
            bintang += 1
        else:
            print("#", end="")
            pagar += 1  

    print("\n* = ", bintang)
    print("# = ", pagar)

    if(bintang >= pagar):
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

bahasa = input()
kode = input()
kode_shikamaru(bahasa, kode)
